### PR TITLE
executor: support more simple statements

### PR DESCRIPTION
**Pull request by coocood**
`DoStmt`, `BeginStmt`, `CommitStmt`, `RollbackStmt`.
`CreateUserStmt`, `SetPwdStmt`.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: executor/executor_simple.go [hunk 1]
```diff
@@ -163,3 +179,101 @@ func (e *SimpleExec) executeSetCharset(s *ast.SetCharsetStmt) error {
...
+	users := make([]string, 0, len(s.Specs))
+	for _, spec := range s.Specs {
+		userName, host := parseUser(spec.User)
+		exists, err1 := userExists(e.ctx, userName, host)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		if exists {
+			if !s.IfNotExists {
+				return errors.Errorf("Duplicate user")
```
Comment (qiuyesuifeng) [inline executor/executor_simple.go]: errors.New

File: executor/executor_simple.go [hunk 2]
```diff
@@ -163,3 +179,101 @@ func (e *SimpleExec) executeSetCharset(s *ast.SetCharsetStmt) error {
...
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// parse user string into username and host
+// root@localhost -> roor, localhost
+func parseUser(user string) (string, string) {
+	strs := strings.Split(user, "@")
+	return strs[0], strs[1]
```
Comment (qiuyesuifeng) [inline executor/executor_simple.go]: Any need to check len(strs)?

Comment (coocood) [inline executor/executor_simple.go]: The argument is always `UserSpec.User` which must contains an '@' character.

**Comment by coocood** [comment]
addressed comment.
@qiuyesuifeng PTAL

File: stmt/stmts/grant.go [hunk 3]
```diff
@@ -538,3 +538,24 @@ func (s *GrantStmt) getTargetSchemaAndTable(ctx context.Context) (*model.DBInfo,
 	}
 	return db, tbl, nil
 }
+
+func userExists(ctx context.Context, name string, host string) (bool, error) {
```
Comment (qiuyesuifeng) [inline stmt/stmts/grant.go]: is there a userExists function already?

Comment (coocood) [inline stmt/stmts/grant.go]: this function was in a deleted file.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**